File: examples/python/langchain_server_side.py
from langchain_hyperspace import HyperspaceVectorStore
from langchain_core.documents import Document

def main():
    """Example using HyperspaceDB server-side vectorization.
    
    No Embeddings function (like OpenAI) is needed here.
    """
    
    # 1. Initialize Store with use_server_side_embedding=True
    vectorstore = HyperspaceVectorStore(
        host="localhost",
        port=50051,
        collection_name="server_side_python_demo",
        use_server_side_embedding=True,
        api_key="I_LOVE_HYPERSPACEDB"
    )

    # 2. Add Documents (server will vectorize them)
    print("Adding documents (server-side embedding)...")
    documents = [
        Document(
            page_content="Hyperbolic geometry is useful for hierarchical data.",
            metadata={"topic": "geometry"}
        ),
        Document(
            page_content="Deep learning models are trained on massive datasets.",
            metadata={"topic": "ai"}
        )
    ]
    ids = vectorstore.add_documents(documents)
    print(f"Added documents with IDs: {ids}")

    # 3. Search (server will vectorize the query)
    print("Searching for 'mathematics'...")
    results = vectorstore.similarity_search("mathematics", k=1)
    if results:
        print(f"Top result: {results[0].page_content}")
        print(f"Metadata: {results[0].metadata}")

    # Success!
    print("Done!")

if __name__ == "__main__":
    main()

File: examples/ts/langchain_server_side.ts
import { HyperspaceStore } from "../../integrations/langchain-js/src/vectorstores";
import { HyperspaceClient } from "hyperspace-sdk-ts";
import { Document } from "@langchain/core/documents";

/**
 * This example demonstrates using HyperspaceDB without any external embedding provider
 * (like OpenAI or HuggingFace). The server handles vectorization automatically.
 */
async function main() {
    // 1. Initialize Client
    const client = new HyperspaceClient("localhost:50051", "I_LOVE_HYPERSPACEDB");
    
    // 2. Initialize Store with useServerSideEmbedding: true
    // Note: We don't provide an 'embeddings' object (passing undefined/dummy)
    const store = new HyperspaceStore(undefined, {
        client,
        collectionName: "server_side_demo",
        useServerSideEmbedding: true,
    });

    // 3. Add Documents (Vectors will be generated on the server)
    console.log("Adding documents (server-side embedding)...");
    await store.addDocuments([
        new Document({ 
            pageContent: "Quantum computing uses qubits instead of bits.",
            metadata: { topic: "physics" } 
        }),
        new Document({ 
            pageContent: "Artificial intelligence is shifting the technology landscape.",
            metadata: { topic: "tech" } 
        })
    ]);

    // 4. Search (Server will also vectorize the query)
    console.log("Searching for 'science stuff'...");
    const results = await store.similaritySearch("science stuff", 1);
    console.log("Result:", results[0].pageContent);
    console.log("Metadata:", results[0].metadata);

    client.close();
    console.log("Done!");
}

main().catch(console.error);
